Index: app/src/main/java/br/unb/unbsolidaria/SignupUserActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/br/unb/unbsolidaria/SignupUserActivity.java	(revision ec28086861927d8a92072fa593a05932f96f0572)
+++ app/src/main/java/br/unb/unbsolidaria/SignupUserActivity.java	(revision )
@@ -5,7 +5,9 @@
 import android.os.Bundle;
 import android.support.v7.app.AppCompatActivity;
 import android.text.Editable;
+import android.text.InputType;
 import android.text.TextWatcher;
+import android.text.InputFilter;
 import android.view.KeyEvent;
 import android.view.View;
 import android.widget.Button;
@@ -72,16 +74,39 @@
 
             @Override
             public void afterTextChanged(Editable s) {
-                if(_cpfText.getText().length() == 9 && !ValidaCadastro.isValidCPF(_cpfText.getText().toString())){
+                if(_cpfText.getText().length() >= 11 && !ValidaCadastro.isValidCPF(_cpfText.getText().toString())){
                     _cpfText.setTextColor(Color.RED);
+                    _cpfText.setError("insira um CPF válido");
                 } else {
                     _cpfText.setTextColor(Color.BLACK);
                 }
             }
         });
+
+        _cepText.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+
-    }
+            }
 
-    @Override
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+                if(_cepText.getText().length() >= 8 && !ValidaCadastro.isValidCEP(_cepText.getText().toString())){
+                    _cepText.setTextColor(Color.RED);
+                    _cepText.setError("insira um CEP válido");
+                } else {
+                    _cepText.setTextColor(Color.BLACK);
+                }
+            }
+        });
+    }
+
+    @Override
     public boolean onKeyDown(int keyCode, KeyEvent event) {
         setResult(RESULT_BACK);
         return super.onKeyDown(keyCode, event);
@@ -154,6 +179,7 @@
         } else {
             _cpfText.setError(null);
         }
+
         if (email.isEmpty() || !android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches()) {
             _emailText.setError("endereço inválido");
             valid = false;
@@ -174,6 +200,7 @@
         } else {
             _matriculaText.setError(null);
         }
+
         if (password.isEmpty() || password.length() < 4 || password.length() > 10) {
             _passwordText.setError("deve ter entre 4 e 10 caracteres");
             valid = false;
